File: Golang/3.go
package main

import (
	"fmt"
)

type ContaBancaria struct {
	titular string
	saldo   float64 
}

func NovaContaBancaria(titular string, saldoInicial float64) *ContaBancaria {
	return &ContaBancaria{
		titular: titular,
		saldo:   saldoInicial,
	}
}


func (c *ContaBancaria) Depositar(valor float64) {
	if valor > 0 {
		c.saldo += valor
		fmt.Printf("Depósito de R$%.2f realizado com sucesso!\n", valor)
	} else {
		fmt.Println("O valor de depósito deve ser positivo.")
	}
}


func (c *ContaBancaria) Sacar(valor float64) {
	if valor > 0 {
		if valor <= c.saldo {
			c.saldo -= valor
			fmt.Printf("Saque de R$%.2f realizado com sucesso!\n", valor)
		} else {
			fmt.Println("Saldo insuficiente.")
		}
	} else {
		fmt.Println("O valor de saque deve ser positivo.")
	}
}


func (c *ContaBancaria) ConsultarSaldo() string {
	return fmt.Sprintf("Saldo atual: R$%.2f", c.saldo)
}

func main() {
	
	conta := NovaContaBancaria("João", 1000)

	
	fmt.Println(conta.ConsultarSaldo())
	conta.Depositar(500)
	fmt.Println(conta.ConsultarSaldo())
	conta.Sacar(300)
	fmt.Println(conta.ConsultarSaldo())
	conta.Sacar(1500) 

File: Python/3.py
class ContaBancaria:
    def __init__(self, titular, saldo_inicial=0):
        self.titular = titular
        self.__saldo = saldo_inicial 


    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f"Depósito de R${valor:.2f} realizado com sucesso!")
        else:
            print("O valor de depósito deve ser positivo.")


    def sacar(self, valor):
        if valor > 0:
            if valor <= self.__saldo:
                self.__saldo -= valor
                print(f"Saque de R${valor:.2f} realizado com sucesso!")
            else:
                print("Saldo insuficiente.")
        else:
            print("O valor de saque deve ser positivo.")

  
    def consultar_saldo(self):
        return f"Saldo atual: R${self.__saldo:.2f}"

  
    @property
    def saldo(self):
        return self.__saldo


conta = ContaBancaria("João", 1000)


print(conta.consultar_saldo())
conta.depositar(500)
print(conta.consultar_saldo())
conta.sacar(300)
print(conta.consultar_saldo())
conta.sacar(1500)  
